Dockerfile

```diff
@@ -76,7 +76,7 @@ COPY pnpm-lock.yaml .
 COPY .npmrc .
 
 # copy the nuxt configuration file
-COPY --from=builder /usr/app/nuxt.config.js .
+COPY --from=builder /usr/app/nuxt.config.ts .
```

Comment:
:scream: For some reason I didn't actually think we could use TypeScript for the nuxt config until Nuxt bridge/Nuxt 3 but this is great!!

Reply:
Question: if the previous stage is the builder and its entrypoint is `npm run build`, does that means that we shouldn't have any `.ts` file at this point?

Reply:
Question: if the previous stage is the `builder` and its entrypoint is `npm run build`, does that means that we shouldn't have any `.ts` file at this point?

Reply:
That's a good question :thinking: The Nuxt config is used both to generate the Webpack config but also to configure some runtime properties for Nuxt, so it's used both for building _and_ for runtime. Nuxt's runtime doesn't use TypeScript though, at least not in Nuxt 2, I don't think.

Reply:
Ok I understand now, I went to the docs, and `npm start` can differ its behavior depending the way you build the application (`npm run build`).

Since the [deployment target](https://nuxtjs.org/docs/features/deployment-targets) is not being specified in the [config file](https://github.com/WordPress/openverse-frontend/blob/main/nuxt.config.js), then the default value is [`server`](https://nuxtjs.org/docs/get-started/commands/#target-server-default-value) which means, we need Node.js to run the application from the generated `.nuxt` folder.

> nuxt start - Start the production server (after running nuxt build). Use it for Node.js hosting like Heroku, Digital Ocean, etc.

That means, at this point, if we want to keep using typescript (after building the application from the previous stage) we will require `ts-node`. Or, my preferred option, as you said Sara, don't convert the config file so it can be read by Node.js runtime without any external dependency.

Reply:
Yeah, I'd prefer if we kept it as TS for now until we move to Nuxt bridge. To use TS for this properly it looks like we'd have to bring in another module for it: https://typescript.nuxtjs.org/guide/runtime/

We can always add type-checking to the file even if it stays as JSDoc though; it just can't consume actual TS files (which means anything we use in the `nuxt.config.js` needs to stay as regular JS with JSDoc types instead of getting converted to TypeScript).

Reply:
If we don't use TS for `nuxt-config`, we cannot import the TypeScript versions of the utils `node-env`, `env` and `sentry-config` in it. I'm not entirely sure why making the `nuxt-config` file TS works but if I change it to JS, it can no longer import any TS files.

Reply:
JS files can't import TS files without the JS files being compiled first. Because a `.js` nuxt.config doesn't get compiled, it won't be able to import TS directly I guess (because it's not importing anything from the compiled bundle).

Considering the e2e tests are passing (they run on the `pnpm build && pnpm start` version of the app, for @rbadillap's context) I guess it's safe.

Also don't know why it's working though (which scares me a little bit :sweat_smile:)